GetAllStreetcodes endpoint. Implemented query for selecting all streetcodes.

Kasterov opened the pull request:
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)


## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

Kasterov committed:
Merge branch 'newFixed' into admin-panel-related-changes

Kasterov committed:
SoftDeletingUtils added

Kasterov committed:
SoftDeleting handler and UpdateStatus endpoints

Kasterov committed:
Some fixes

Delete useless usings

Kasterov committed:
Back how it was

Back how it was week ago

Kasterov committed:
GetAll with query implemented

Yea. A lot of work.
1. GetAllStreetcodesRequest added to representation layer.

2. Change query for Mediatr Hadnler GetAllStreetcodes.

3. Fix deleted handler with throwing exeption.

Yeah, that a lot... Sry. Never gona do it again.

Kasterov committed:
Merge branch 'newFixed' into admin-panel-related-changes

oshut commented on Streetcode/Streetcode.BLL/MediatR/Streetcode/Streetcode/Delete/DeleteStreetcodeHandler.cs:
```diff
@@ -25,6 +30,12 @@ public async Task<Result<Unit>> Handle(DeleteStreetcodeCommand request, Cancella
         var resultIsSuccess = await _repositoryWrapper.SaveChangesAsync() > 0;
-        return resultIsSuccess ? Result.Ok(Unit.Value) : Result.Fail(new Error("Failed to delete a streetcode"));
+
+        if (!resultIsSuccess)
```
Is there any point of refactoring it in this way?
also it'd would be better to name this variable as `isSuccessResult` IMHO

oshut replied:
if it's common convention, it's okay with me

oshut replied:
but I see you've approved this PR with ternary operator in similiar place - https://github.com/ita-social-projects/StreetCode/pull/132/files#diff-a5606f5d4d156694e3b38579aa50412b5b3e30bbd53d977626ba3f8b4624b6d2R31

So are we really talking about some conventions here?

oshut commented on Streetcode/Streetcode.BLL/MediatR/Streetcode/Streetcode/GetAll/GetAllStreetcodesHandler.cs:
```diff
@@ -20,13 +22,67 @@ public GetAllStreetcodesHandler(IRepositoryWrapper repositoryWrapper, IMapper ma
+            return Result.Fail(new Error($"Cannot find any streetcodes"));
         }
 
+        if (request.Title is not null)
```
I'm little bit new to this repo and it's code, but don't you think that this method are too big and it can be optimized by splitting it to private methods?

Kasterov replied:
If you are talking about whole GetAllStreetcodes endpoint, It can be done this way, but this logic, I suppose, has to be used just in current endpoint, in oposite way, I prefer to create service and inject it, maybe it will be useful in other parts of project.

oshut replied:
splitting one big method in few small it's not only about reusability, but about readability and it makes it easy to understand

Kasterov replied:
Agree with you. And I suposse it is good idea. If logic of endpoint will be approved by Teachlead, I could change it that way (sepparate logic for different private methods).

oshut replied:
It was just suggesting, so final decision is up to you)

Tysyatsky reviewed (commented):
Good job, just small remarks and it will be perfect.

Kasterov committed:
ConsoleWriteline deleted

Delete unnecessary Console.Writeline() methods

Kasterov committed:
Merge branch 'admin-panel-related-changes' of https://github.com/ita-social-projects/StreetCode into admin-panel-related-changes

Kasterov committed:
Ternar operation changes

Now instead of "if" such handlers as "DeleteHandler", "SoftDeleteHandler" and "UpdateStatus" use ternar operation for final checking of success.

Kasterov committed:
Handler private methods

For getAllStreetcodes Handler implemented static methods

Kasterov committed:
IQuariable implemented

Now instead of IEnumarable selection from Db we use IQuariable

Kasterov committed:
Response GetAllStreetcodes apdated

Now this request returns now only list of streetcodes but also amount of possible pages according to pagination request

Kasterov committed:
Fixes

Filtering fixes. Now everything works properly.
